File: 5.1.py
def pgcd(a,b):
    while b!=0:
        m=b
        b=a%b
        a=m
    return a
a = int(input("Enter the first number: "))
b = int(input("Enter the second number: "))
print(f"GCD is: {pgcd(a, b)}")

File: 5.1.c
#include<stdio.h>
int pgcd(int a,int b){
    while(b!=0){
        int m=b;
        b=a%b;
       a=m;
    }
    return a;
}
int main(){
    int a,b;
    printf("enter 2 numbers \n");
    scanf("%d %d",&a,&b);
    printf("GCD is : \n  %d ",pgcd(a,b ));
    return 0;

}